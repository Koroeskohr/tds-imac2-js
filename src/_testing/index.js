/* eslint-disable import/prefer-default-export */
import chalk from 'chalk';

export const testWithErrorMessage = (test, message) => {
  try {
    test();
  } catch (e) {
    const err = new Error(
      `${chalk.blue(chalk.bold('Teacher\'s hint >'))} ${message}. \n\n${e}`,
    );
    Error.captureStackTrace(err, testWithErrorMessage);
    throw err;
  }
};
