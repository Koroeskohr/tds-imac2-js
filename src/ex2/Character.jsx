const Character = () => null;

export default Character;
