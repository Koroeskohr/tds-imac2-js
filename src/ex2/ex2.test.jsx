import React from 'react';
import { render, screen } from '@testing-library/react';
import { ExerciseTwoPage, character } from '.';
import { testWithErrorMessage } from '../_testing';
import Character from './Character';

describe('Part 1', () => {
  test('The page should be intact', () => {
    render(<ExerciseTwoPage />);
    testWithErrorMessage(() => {
      const el = screen.queryByText('Here are our characters');
      expect(el).not.toBeNull();
    }, 'The initial p tag is gone, you might not return anything');
  });

  test('Character should be defined', () => {
    const {
      age, isInPrison, job, name, picture,
    } = character;
    const c = (
      <Character
        name={name}
        age={age}
        job={job}
        imageUrl={picture}
        isInPrison={isInPrison}
      />
    );
    testWithErrorMessage(() => {
      expect(c).not.toBeNull();
    }, 'Character is not defined yet');
  });

  test('Character does not contain an undefined value', () => {
    const {
      age, isInPrison, job, name, picture,
    } = character;
    render(
      <Character
        name={name}
        age={age}
        job={job}
        imageUrl={picture}
        isInPrison={isInPrison}
      />,
    );
    testWithErrorMessage(() => {
      expect(screen.queryByText('undefined')).toBeNull();
    }, 'The content of your component contains undefined, this should not happen');
  });

  test('Structure is valid', () => {
    const {
      age, isInPrison, job, name, picture,
    } = character;
    const { container } = render(
      <Character
        name={name}
        age={age}
        job={job}
        imageUrl={picture}
        isInPrison={isInPrison}
      />,
    );
    testWithErrorMessage(() => {
      expect(container).toMatchSnapshot();
    }, 'The content of the Character component is invalid');
  });

  test('Page contains the Character component', () => {
    render(<ExerciseTwoPage />);
    testWithErrorMessage(() => {
      expect(screen.queryByText('Andy Dufresne')).not.toBeNull();
    }, 'The Character component with Andy Dufresne is not in the page');
  });
});

describe('Part 2', () => {
  test.skip('No test here', () => {});
});
