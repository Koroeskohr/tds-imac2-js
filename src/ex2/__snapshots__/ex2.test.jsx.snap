// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Part 1 Structure is valid 1`] = `
<div>
  <div>
    <h1>
      Andy Dufresne
    </h1>
    <h2>
      Banker
    </h2>
    <p>
      35
    </p>
    <p>
      Is in prison
    </p>
  </div>
</div>
`;
